protos/flyteidl/artifact/artifacts.proto

```diff
@@ -0,0 +1,196 @@
+syntax = "proto3";
+package flyteidl.artifact;
+
+option go_package = "github.com/flyteorg/flyteidl/gen/pb-go/flyteidl/artifact";
+
+import "flyteidl/core/literals.proto";
+import "flyteidl/core/types.proto";
+import "flyteidl/core/identifier.proto";
+
+
+message ArtifactKey {
+  // Project and domain as we're all used to. Just to align with existing uniqueness constructs.
+  string project = 1;
+  string domain = 2;
+
+  // name can be auto-generated by the system in the case of natural outputs or un-named uploads
+  // or manually named by the user, if they care to.
+  // Please see the confluence doc for more discussion, this is one of the trickier bits of the design.
+  // Maybe we should always have an auto-generated string here and the user-facing "name" is a different field.
+  string name = 3;
+}
+
+message ArtifactID {
```

Comment:
Having both an ArtifactKey and an ArtifactID is a bit confusing...

Reply:
@kumare3 wanted this :)